### PR TITLE
*: add specified columns for LOAD DATA INFILE Syntax

### PR DESCRIPTION
**Pull request by GregoryIan**
@zimulala @tiancaiamao PTAL

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -280,17 +280,25 @@ func (b *executorBuilder) buildLoadData(v *plan.LoadData) Executor {
 		b.err = errors.Errorf("Can not get table %d", v.Table.TableInfo.ID)
 		return nil
 	}
+	insertVal := &InsertValues{ctx: b.ctx, Table: tbl, Columns: v.Columns}
+	tableCols := tbl.Cols()
```
Comment (shenli) [inline executor/builder.go]: Should call tbl.WritableCols().

Comment (GregoryIan) [inline executor/builder.go]: yep, forgot column' state

**Comment by hanfei1991** [comment]
If you use a wrong column name, will you report an error?

**Comment by GregoryIan** [comment]
yep, It would use [table.FindCols()](https://github.com/pingcap/tidb/blob/c1796ac66901867d403ec7daddad914c88364372/executor/write.go#L740) to check columns

**Comment by hanfei1991** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@zimulala Please take a look at your spare time.

File: executor/write_test.go [hunk 2]
```diff
@@ -805,15 +806,48 @@ func (s *testSuite) TestLoadDataEscape(c *C) {
...
+// reuse TestLoadDataEscape's test case :-)
+func (s *testSuite) TestLoadDataSpecifiedCoumns(c *C) {
+	defer func() {
+		s.cleanEnv(c)
+		testleak.AfterTest(c)()
+	}()
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test; drop table if exists load_data_test;")
+	tk.MustExec(`create table load_data_test (id int PRIMARY KEY AUTO_INCREMENT, c1 int, c2 varchar(255) default "def", c3 int default 0);`)
+	tk.MustExec("load data local infile '/tmp/nonexistence.csv' into table load_data_test")
```
Comment (zimulala) [inline executor/write_test.go]: Use `load data local infile '/tmp/nonexistence.csv' into table load_data_test (c1,c2) ` instead of it or remove it .

**Comment by zimulala** [comment]
If the "load data" operation specified columnA, because the data is large, we need to do a lot of transactions.
When doing transactions, someone changed the schema, columnA state is not "WriteOnly", its state becomes "DeleteOnly". Do we need to do some processing?

**Comment by GregoryIan** [comment]
yep, such case would cause data corruption.  If we need to deal with  it, we should defer these schema changes or load newest schema in every transactions of `load data` then it would return unknown columnA, it maybe confuse user.  what's your thoughts about it while you achieve `load data` , some new ideas now?  @zimulala are there some convenient way that TiDB provide but I don't know?

**Comment by zimulala** [comment]
Can we check the schema information in each transaction？ If the columns' schema information changes, we exit the current transaction and return warning information? @GregoryIan

File: session.go [hunk 3]
```diff
@@ -1067,6 +1067,15 @@ func (s *session) prepareTxnCtx() {
 	}
 }
 
+// RefreshTxnCtx implements context.RefreshTxnCtx interface.
+func (s *session) RefreshTxnCtx() error {
```
Comment (zimulala) [inline session.go]: Why implements this function? I feel like this function like `NewTxn`.

**Comment by GregoryIan** [comment]
@shenli @zimulala PTAL

File: context/context.go [hunk 4]
```diff
@@ -51,6 +51,11 @@ type Context interface {
 
 	GetSessionManager() util.SessionManager
 
+	// RefreshTxnCtx commit old transaction without retry,
```
Comment (zimulala) [inline context/context.go]: s/commit/commits

File: context/context.go [hunk 5]
```diff
@@ -51,6 +51,11 @@ type Context interface {
...
+	// and create a new transation.
```
Comment (zimulala) [inline context/context.go]: s/create/creates

**Comment by zimulala** [comment]
LGTM

**Pull request merged**